packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeFormatService.ts

```diff
@@ -0,0 +1,566 @@
+    }
+
+    const requestPresentation = options.attachment.getDataAsJson<RequestPresentationExchangeOptions>()
+
+    const credential: IVerifiableCredential = options.proofFormats.presentationExchange.formats
+
+    const pex: PEXv1 = new PEXv1()
+
+    const subject: ICredentialSubject = credential.credentialSubject as ICredentialSubject
+
+    // Credential is allowed to be presented without a subject id. In that case we can't prove ownerhsip of credential
+    // And it is more like a bearer token.
+    // In the future we can first check the holder key and if it exists we can use that as the one that should authenticate
+    // https://www.w3.org/TR/vc-data-model/#example-a-credential-issued-to-a-holder-who-is-not-the-only-subject-of-the-credential-who-has-no-relationship-with-the-subject-of-the-credential-but-who-has-a-relationship-with-the-issuer
+    if (!subject?.id) {
+      throw new AriesFrameworkError(
+        'Credential subject missing from the selected credential for creating presentation.'
+      )
+    }
+
+    const didResolutionResult = await this.didResolver.resolve(agentContext, subject.id)
+
+    if (!didResolutionResult.didDocument) {
+      throw new AriesFrameworkError(`No did document found for did ${subject.id}`)
+    }
+
+    if (!didResolutionResult.didDocument?.authentication) {
+      throw new AriesFrameworkError(`No did authentication found for did ${subject.id} in did document`)
+    }
+
+    if (!didResolutionResult.didDocument?.verificationMethod) {
+      throw new AriesFrameworkError(`No did verification method found for did ${subject.id} in did document`)
+    }
+
+    const proofPurpose = IProofPurpose.authentication
+
+    // the signature suite to use for the presentation is dependant on the credentials we share.
+
+    // assertionMethod?: Array<string | VerificationMethod>
+
+    // 1. Get the key for this given proof purpose in this DID document
+    const keyId = didResolutionResult.didDocument[proofPurpose] as string[]
+
+    // get keys from the did document section containing the proof purpose
+
+    // 2. Map the Key Id to a key. Key contains publicKey and keyType attributes
+    const privateKey: Key = keyReferenceToKey(didResolutionResult.didDocument, keyId[0])
+
+    // 3. Use the retrieved key to determine proof type
+    const proofType = keyTypeToProofType(privateKey)
+
+    if (!proofType) {
+      throw new AriesFrameworkError(`Unsupported key type: ${privateKey.keyType}`)
+    }
+
+    const params: PresentationSignOptions = {
+      holder: subject.id,
+      proofOptions: {
+        type: proofType,
+        proofPurpose: IProofPurpose.assertionMethod,
+        challenge: requestPresentation.options?.challenge,
+      },
+      signatureOptions: {
+        verificationMethod: (didResolutionResult.didDocument?.authentication[0]).toString(),
+        keyEncoding: KeyEncoding.Base58,
+        privateKey: privateKey.publicKeyBase58,
```

Comment:
This is not needed I think?

```suggestion
```

It's also wrong as we're passing a public key to a privateKey property